src/main/java/com/echobox/api/linkedin/client/DefaultWebRequestor.java

```diff
@@ -82,361 +55,138 @@ public class DefaultWebRequestor implements WebRequestor {
-          @Override
-          public void initialize(HttpRequest request) throws IOException {
-            credential.initialize(request);
-            request.setParser(new JsonObjectParser(JSON_FACTORY));
-          }
-        });
-
-    return requestFactory;
+        .connectTimeout(Duration.ofMillis(connectTimeout)).build();
+    this.readTimeout = readTimeout;
   }
-
+  
   @Override
   public Response executeGet(String url) throws IOException {
     return executeGet(url, null);
   }
   
   @Override
   public Response executeGet(String url, Map<String, String> headers) throws IOException {
-    return execute(url, null, java.net.http.HttpRequest.Builder::GET, headers);
+    return execute(url, null, HttpRequest.Builder::GET, headers);
   }
-
+  
   @Override
   public Response executePost(String url, String parameters, String jsonBody) throws IOException {
-    return executePost(url, parameters, jsonBody, null, new BinaryAttachment[0]);
+    return executePost(url, parameters, jsonBody, null);
   }
-
-  // CPD-OFF
+  
   @Override
   public Response executePost(String url, String parameters, String jsonBody,
-      Map<String, String> headers,
-      BinaryAttachment... binaryAttachments)
-      throws IOException {
-    
-    if (binaryAttachments == null) {
-      binaryAttachments = new BinaryAttachment[0];
-    }
-
-    try {
-      GenericUrl genericUrl = getGenericURL(url, parameters);
-
-      HttpRequest request;
-      HttpHeaders httpHeaders = new HttpHeaders();
-  
-      // If we have binary attachments, the body is just the attachments and the
-      // other parameters are passed in via the URL.
-      // Otherwise the body is the URL parameter string.
-      if (binaryAttachments.length > 0) {
-        // Set the media type
-        MultipartContent content = new MultipartContent()
-            .setMediaType(new HttpMediaType("multipart/form-data")
-                .setParameter("boundary", MULTIPART_BOUNDARY));
-        
-        for (BinaryAttachment binaryAttachment : binaryAttachments) {
-          MultipartContent.Part part = new MultipartContent.Part(
-              new ByteArrayContent(binaryAttachment.getContentType(), binaryAttachment.getData()));
-          HttpHeaders set = new HttpHeaders().set("Content-Disposition",
-              format("form-data; name=\"%s\"; filename=\"%s\"",
-                  createFormFieldName(binaryAttachment), binaryAttachment.getFilename()));
-  
-          part.setHeaders(set);
-          content.addPart(part);
-  
-          httpHeaders.set("Connection", "Keep-Alive");
-        }
-        
-        request = requestFactory.buildPostRequest(genericUrl, content);
-      } else {
-        if (jsonBody != null) {
-          request = requestFactory.buildPostRequest(genericUrl, getJsonHttpContent(jsonBody));
-
-          // Ensure the response headers are also set to JSON
-          request.setResponseHeaders(new HttpHeaders().set(FORMAT_HEADER, "json"));
```

Comment:
I wasn't sure how to add this for the response header.

Reply:
Not even sure what it means for a request to set a response header, if this works without it then all good